### PR TITLE
Doc updates for macOS 14.5 Sonoma

### LINKED ISSUE #1123: Document the procedure for installing a specific version of Command Line Tools on the Mac
**Is your feature request related to a problem? Please describe.**
This is needed now to address the issue that the lastest command line tools (15.3) does not work for spack-stack builds on MacOS, but the command line tools version 15.1 does work. Users need to know how to back up to the 15.1 version of the command line tools.

**Describe the solution you'd like**
Have the above procedure documented for spack-stack users.

**Additional context**
None

### PR DESCRIPTION
**Pull request by eap**
This change updates the documentation to have additional information on installing spack-stack in a macOS 14.5 Sonoma environment. Some additional helpful info has been added on selecting a working clang compiler and several other smaller updates were added.

## Issues

Fixes: https://github.com/JCSDA/spack-stack/issues/1123

### CODE REVIEW

File: doc/source/NewSiteConfigs.rst [hunk 1]
```diff
@@ -93,6 +93,9 @@ These instructions are meant to be a reference that users can follow to set up t
 .. note::
    If you encounter build errors for gdal later on in spack-stack (see :numref:`Section %s <KnownIssues>`), you may need to install the full ``Xcode`` application and then switch ``xcode-select`` over with ``sudo xcode-select -s /Applications/Xcode.app/Contents/Developer`` (change the path if you installed Xcode somewhere else).
 
+.. note::
+   I you have clang 15.x, please read the Known Issues entry on clang 15.x (see :numref:`Section %s <KnownIssues>`).
```
Comment (eap) [inline doc/source/NewSiteConfigs.rst]: s/I you have/If you have/. This is fixed locally but I'll push the change after further reviews.

**Review by climbfuji — APPROVED** [review]

File: doc/source/KnownIssues.rst [hunk 2]
```diff
@@ -123,9 +123,9 @@ macOS
...
 
-6. ``apple-clang@15.0.0`` not yet supported
+6. Not all versions of ``apple-clang@15.0.0`` are currently supported.
 
-   Building with ``apple-clang@15.0.0`` is under development and should be working soon. In the meantime, please use ``apple-clang@14.x`` or older versions.
+   If you are running macOS Sonoma 14.5 you may have a recent version of clang that does not yet build spack packages. On the console you should run ``clang --version`` and verify that the underlying clang compiler is ``clang-1500.1.0.2.5`` which is the version provided with Command Line Tools for XCode 15.1. You can downgrade your Command Line Tool packages by first running ``sudo rm -rf /Library/Developer/CommandLineTools`` then you can download and install the Command Line Tools 15.1 APK from `developer.apple.com <https://developer.apple.com/download/all/>_`. The Apple Developer site requires an iCloud login.
```
Comment (srherbener) [inline doc/source/KnownIssues.rst]: The link to the developer.apple.com download/all web page didn't render properly. I think the trailing quote and underscore are transposed.

```suggestion
   If you are running macOS Sonoma 14.5 you may have a recent version of clang that does not yet build spack packages. On the console you should run ``clang --version`` and verify that the underlying clang compiler is ``clang-1500.1.0.2.5`` which is the version provided with Command Line Tools for XCode 15.1. You can downgrade your Command Line Tool packages by first running ``sudo rm -rf /Library/Developer/CommandLineTools`` then you can download and install the Command Line Tools 15.1 APK from `developer.apple.com <https://developer.apple.com/download/all/>`_. The Apple Developer site requires an iCloud login.
```

File: doc/source/NewSiteConfigs.rst [hunk 3]
```diff
@@ -226,20 +229,23 @@ Remember to activate the ``lua`` module environment and have MacTeX in your sear
    spack external find --scope system \
        --exclude bison --exclude openssl \
        --exclude python --exclude gettext
-   spack external find --scope system libiconv
```
Comment (srherbener) [inline doc/source/NewSiteConfigs.rst:229]: WIth this change, here and line 241 below, do we also need to add `brew install libiconv` to the brew commands listed above (around line 141)?

**Comment by barlage** [comment]
If anyone's interested in my personal experience installing on macOS 14.5 Sonoma recently, here are my notes from running the commands in section 6.1.4-4. I did not `brew install libiconv` or need to externally find it. Also, perl seems to have been found from the first find. Not a spack power user, just some information if it's useful.

```
spack external find --scope system --exclude bison --exclude openssl --exclude python --exclude gettext

==> The following specs have been detected on this system and added to /Users/michael.barlage/spack-stack/envs/ufs.macos/site/packages.yaml
cmake@3.29.6   curl@8.6.0  git@2.39.3  git-lfs@3.5.1  m4@1.4.6       perl@5.30.3
coreutils@9.5  flex@2.6.4  git@2.45.2  gmake@3.81     openssh@9.6p1  perl@5.34.1

spack external find --scope system libiconv

==> No new external packages detected

spack external find --scope system perl

==> No new external packages detected

spack external find --scope system wget

==> The following specs have been detected on this system and added to /Users/michael.barlage/spack-stack/envs/ufs.macos/site/packages.yaml
wget@1.24.5

PATH="$HOMEBREW_ROOT/opt/curl/bin:$PATH" spack external find --scope system curl

==> The following specs have been detected on this system and added to /Users/michael.barlage/spack-stack/envs/ufs.macos/site/packages.yaml
curl@8.8.0

PATH="$HOMEBREW_ROOT/opt/qt@5/bin:$PATH" spack external find --scope system qt

==> The following specs have been detected on this system and added to /Users/michael.barlage/spack-stack/envs/ufs.macos/site/packages.yaml
qt@5.15.13
```

File: doc/source/NewSiteConfigs.rst (hunk 3 above)
Comment (eap) [reply doc/source/NewSiteConfigs.rst:229]: done

**Review by srherbener — APPROVED** [review]
Thanks for addressing my comments!

**Comment by eap** [comment]
I've added some additional ignore directives to the CI workflows in preparation for a fix that should resolve the issue of missing required tests when changes are made to ignored directories (like docs or site configs). As an added benefit, this change triggered the required tests allowing the "notest" workflow variants to be submitted in a dedicated PR allowing further review and discussion on the change.

**Pull request merged**